components/support/sql/src/conn_ext.rs

```diff
@@ -213,3 +217,72 @@ impl<'conn> ConnExt for UncheckedTransaction<'conn> {
         &*self
     }
 }
+
+/// This transaction is suitable for when a transaction is being used only
+/// for performance when updating a larger number of rows (ie, when the
+/// transaction is not necessary for data integrity). You can specify a
+/// duration for the maximum amount of time the transaction should be held.
+/// You should regularly call .maybe_commit() as part of your processing, and
+/// if the current transaction has been open for greater than the specified
+/// duration the transaction will be committed and another one started.
+/// You should always call .commit() at the end. Note that if you call
+/// .rollback() at the end, only the current transaction will be rolled back.
```

Comment:
On the one hand, I could see this being surprising, and maybe we shouldn't expose a way to roll the transaction back explicitly. On the other, SQLite might roll it back implicitly (due to a constraint violation, invalid SQL, app crash, power loss, interrupt, etc.), so we don't gain much by hiding it. The warning about "don't call `maybe_commit` unless you're really ready to commit" is probably enough.